decred/tests/unit/dcr/test_txscript.py

```diff
@@ -2834,57 +2800,53 @@ def __init__(
                     "303963353933633365393166656465373032392102323364643432643"
                     "235363339643338613663663530616234636434340a00000053ae"
                 ),
-                addrs=[],
-                exception="isn't on secp256k1 curve",
-            )
-        )
-        tests.append(
-            test(
+                exception=DecredError,
+            ),
+            dict(
                 name="empty script",
                 script=ByteArray(b""),
                 addrs=[],
                 reqSigs=0,
-                exception="unsupported script",
-            )
-        )
-        tests.append(
-            test(
+                scriptClass=txscript.NonStandardTy,
+            ),
+            dict(
                 name="script that does not parse",
                 script=ByteArray([opcode.OP_DATA_45]),
                 addrs=[],
                 reqSigs=0,
-                exception="unsupported script",
-            )
-        )
+                scriptClass=txscript.NonStandardTy,
+            ),
+        ]
 
         def checkAddrs(a, b, name):
-            if len(a) != len(b):
-                t.fail(
-                    "extracted address length mismatch. expected %d, got %d"
-                    % (len(a), len(b))
-                )
+            assert len(a) == len(b), (
+                f"extracted address length mismatch. expected %d, got %d"
```

Comment:
You have an `f` letter in front of the string and that means new-style interpolation, but that uses curly brackets, not old-style percent signs.